lib/utils/ember.js

```diff
@@ -142,24 +149,42 @@ function isReopenObject(node) {
   return node.callee.property && node.callee.property.name === 'reopen';
 }
 
-function isEmberComponent(node, filePath) {
-  return isEmberCoreModule(node, 'Component', filePath);
+function isEmberCoreModule(context, node, moduleName) {
+  if (types.isCallExpression(node)) {
+    // "classic" class pattern
+    return isClassicEmberCoreModule(node, moduleName, context.getFilename());
+  } else if (types.isClassDeclaration(node)) {
+    // native classes
+    if (!node.superClass) {
+      return false;
+    }
+    const superClassImportPath = utils.getImportModule(context, node.superClass);
+
+    if (superClassImportPath === CORE_MODULE_IMPORT_PATHS[moduleName]) {
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
Can you add:
```
else {
  assert('Function should only be called on a `CallExpression` (classic class) or `ClassDeclaration` (native class)');
}
```

Reply:
Absolutely

Reply:
I might be totally missing something, but I can't see any `assert` function either in the utils or packages. Did you want to import a specific one?

Reply:
I think that Node has a built-in `assert` helper 

```
const assert = require('assert');
```

Maybe that's the one?

Reply:
Ahhh that's probably it! Will add it now